src/nrniv/nrnste.h

```diff
@@ -1,65 +1,55 @@
-#ifndef nrnste_h
-#define nrnste_h
+#pragma once
+#include <vector>
 // StateTransitionEvent is a finite state machine in which a transtion occurs
 // when the transition condition is true. For speed the transition condition
 // is of the form *var1 > *var2 and allows second order interpolation.
 // Transition conditions are checked only if the source state is the
 // current state.
 
 class HocCommand;
-class StateTransitionEvent;
+struct StateTransitionEvent;
 class STECondition;
 
-class STETransition {
-  public:
-    STETransition();
-    virtual ~STETransition();
```

Comment:
for clarity - as the virtual destructor is removed, I assume there are no cases of classic memory leaks due to a base class pointer pointing to a derived class?